include/seahorn/Expr/ExprMemMap.hh

```diff
@@ -0,0 +1,174 @@
+/// Pointer Expr -> Symbolic Memory Map
```

Comment:
Is this new code or factored out from `HexDump`?

Reply:
factored out from `HexDump`; cut short much of the layers of old design;
now its just:
`ExprMemMap` has `MemCellSet` and `MemExprVisitor`;
`MemCellSet` contains a set of `MemCell`s and functions for access